### ISSUE TITLE
Improve  unicorn/prefer-string-replace-all rule

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/oxc#18967]
The rule must be possible to format all once time.

Consider the following code:

```ts
// Single quote will break @vue/compiler-sfc
export const stringifyProp = (data: unknown): string =>
  JSON.stringify(data).replace(/'/g, '&#39')
```

It will be converted to the following in the first `oxlint --fix`

```ts
// Single quote will break @vue/compiler-sfc
export const stringifyProp = (data: unknown): string =>
  JSON.stringify(data).replaceAll(/'/g, '&#39')
```

Then format to the final result in NEXT `oxlint --fix`

```ts
// Single quote will break @vue/compiler-sfc
export const stringifyProp = (data: unknown): string =>
  JSON.stringify(data).replaceAll("'", '&#39')
```

Since both fixes are triggered by the same rule `unicorn/prefer-string-replace-all`, technically speaking, there should be no blocking things that stop the rule from doing the job once in all. 

This is annoying for linting in Git hooks with lint-staged, as code that passes `oxlint -fix` is still problematic in the next turn.

> code that passes `oxlint -fix` is still problematic in the next turn.

Under this sight, this should be a bug
